source/js/foundation/template-js-handler/continuous-scrolling-box.js

```diff
@@ -0,0 +1,112 @@
+import { gsap } from "gsap";
+import { ScrollTrigger } from "gsap/ScrollTrigger";
+
+export default () => {
+  /**
+   * Creates slideup intro animation when entering a new article. If history.state indicates
+   * that it is coming from another article via scrolltrigger, then we animate the white box that is overlaying the article.
+   */
+  function triggerIntroTransition() {
+    if (sessionStorage.contScrolling) {
+      gsap.to(".page-enter-transition", {
+        y: "-100vh",
+        delay: 0.5,
+        duration: 0.5,
+      });
+      sessionStorage.removeItem("contScrolling");
+    } else if (document.querySelector(".page-enter-transition")) {
+      document.querySelector(".page-enter-transition").remove();
+    }
+  }
+
+  /**
+   * Creates a scroll trigger that will trigger the exit animation when the user scrolls past the bottom of the page.
+   */
+  function initScrollExitTransition() {
+    gsap.to(".cont-scrolling .tw-container", {
+      scrollTrigger: {
+        trigger: ".cont-scrolling",
+        scrub: true,
+        start: "+250 bottom",
+        end: "bottom bottom",
+        toggleActions: "play none none reverse",
+        onLeave: () => {
+          document
+            .querySelector(".cont-scrolling .tw-container")
+            .classList.add("tw-hidden");
+        },
+      },
+      autoAlpha: 0,
+    });
+
+    ScrollTrigger.create({
+      trigger: ".cont-scrolling",
+      start: "top center",
+      end: "bottom bottom",
+      scrub: true,
+      toggleActions: "play none none reverse",
+      onLeave: () => {
+        const outro = gsap.timeline({
+          onComplete: () => {
+            // If it takes more then 400ms show the loading indicator
+            setTimeout(() => {
+              if (document.querySelector(".article-loading")) {
+                document
+                  .querySelector(".article-loading")
+                  .classList.toggle("tw-hidden");
+              }
```

Comment:
Better to make sure not to repeat queries, so this can be rewritten to something like:
```suggestion
              const loadIndicator = document.querySelector(".article-loading");
              if (loadIndicator) {
                loadIndicator.classList.toggle("tw-hidden");
              }
```